context/index.d.ts

```diff
@@ -0,0 +1,17 @@
+import type { ReadableAtom } from '../atom'
+
+// This isn't a part of the public API for now
+type Context = unknown
+
+export type ContextGetter = <T extends ReadableAtom>(store: T) => T
+export type WithCtx = {
+  ctx: ContextGetter
+}
+
+export declare const globalContext: Context
+
+export function createContext(name: string): Context
+export function resetContext(name?: string): void
+export function getContext(name: string): Context | undefined
```

Comment:
Why do we need context name? Can we just use `id` from `createContext`?

Reply:
Yep, that's the same id :) Well, it's called `name` for now, but I'm open to changing it to `id`, that seems even better.